ACCUMULO-4528 Accumulo scripts improvements

mikewalch opened the pull request:
* Accumulo now determines hostname in Java
* Minimized use of environment variables
* Consolidated scripts by moving service.sh and cluster.sh into
  accumulo-service and accumulo-cluster which were moved to contrib/.
  Also, moved masters, tservers, tracers, etc files to contrib/conf/
* Removed accumulo-watcher script as restarting services should not be
  handled by Accumulo scripts
* NUMA and multiple tservers are no longer configured in scripts but
  could be NUMA commands could be added using ACCUMULO_JAVA_PREFIX env
  variable.
* Renamed accumulo-env.sh to accumulo.conf
* Moved create-config and build-native from accumulo script to
  accumulo-config and accumulo-native scripts in contrib/

milleruntime commented on INSTALL.md:
```diff
@@ -57,15 +57,15 @@ The script will ask you questions about your set up. Below are some suggestions:
 Accumulo coordination and worker processes can only communicate with each other
 if they share the same secret key.  To change the secret key set
-`instance.secret` in `conf/accumulo-site.xml`.  Changing this secret key from
+`instance.secret` in `accumulo-site.xml`.  Changing this secret key from
```
Did you want to move accumulo-site.xml out of conf?

mikewalch replied:
It's still located in conf for the tarball distribution but I changed the manual to not include directory/path of the file as accumulo-site.xml could be in other locations like `/etc/accumulo` in downstream packaging.

milleruntime replied:
Got it.  I took the changes out of context.

ctubbsii commented on assemble/contrib/tool.sh:
```diff
@@ -26,14 +26,21 @@ contrib=$( cd -P "$( dirname "$SOURCE" )" && pwd )
-source "$basedir"/libexec/load-env.sh
+conf="${basedir}/conf"
+if [ ! -f "${conf}/accumulo.conf" ]; then
+  echo "accumulo.conf must exist in $conf"
```
Why wouldn't this file be optional?

mikewalch replied:
I made it optional in 7f9da9f12ce5940

ctubbsii commented on assemble/contrib/tool.sh:
```diff
@@ -26,14 +26,21 @@ contrib=$( cd -P "$( dirname "$SOURCE" )" && pwd )
 
 if [[ -z "$HADOOP_PREFIX" ]] ; then
-   echo "HADOOP_PREFIX is not set.  Please make sure it's set globally or in conf/accumulo-env.sh"
+   echo "HADOOP_PREFIX is not set.  Please make sure it's set globally or in $conf/accumulo.conf"
```
I think this is currently necessary for the property interpolation in the `accumulo-site.xml` file, but I'd really like to phase out having it as a requirement.

keith-turner commented on INSTALL.md:
```diff
@@ -25,11 +25,11 @@ source code.  Unpack as follows.
-2. `accumulo-service` - Runs Accumulo processes as services
-3. `accumulo-cluster` - Manages Accumulo cluster on a single node or several nodes
+1. `bin/accumulo` - Runs Accumulo command-line tools and starts Accumulo processes
+2. `contrib/accumulo-service` - Runs Accumulo processes as services
```
What does it mean for a script to be in contrib vs bin?

ctubbsii replied:
In my view, `contrib/` should be for add-ons/wrappers which could be separate projects on their own, but which we are currently maintaining to provide convenient functionality.

I think it makes sense to have a basic launch script in `bin/`, which minimally wraps the execution of java. That minimal wrapper wouldn't make sense as its own project and thus (by my opinion) shouldn't be in `contrib/`.

I'm far less opinionated on the convenience scripts for producing the example configs and for executing `make` on the unpacked native tarball. They could probably go either place.

mikewalch replied:
I thought about having a contrib and a bin directory. I think there should only be a bin/ directory.  Two directories will confuse people.  We can communicate that scripts are optional/add-on in the documentation rather than the directory structure.  To limit the number of scripts in bin, we can combine many of the scripts in contrib into one or two scripts.  If a script doesn't work or we don't want to support it any more the in the tarball distribution, it should either be removed or move to a contrib directory in the repo that doesn't get added to the distribution. Contrib scripts should not be distributed.  As for the scripts in bin/, downstream packagers can pick and choose what they include.

ctubbsii replied:
Seems reasonable.

milleruntime commented:
+1 On the renaming of accumulo-env.sh to accumulo.conf.  The .conf extension is pretty clear what the file is for, its simply for configuration purposes.  The -env.sh extension tells me this shell script is setting up the environment but could be doing more then configuration, which makes me nervous just seeing the name of the file.  From a user perspective, I would be much more inclined and comfortable to modify a .conf file.

mikewalch commented on INSTALL.md:
```diff
@@ -40,13 +40,13 @@ Accumulo has some optional native code that improves its performance and
 with the following command.
 
-    accumulo build-native
+    ./contrib/accumulo-native
```
I pushed a commit where I moved the command to `bin/accumulo-util build-native`

mikewalch commented on INSTALL.md:
```diff
@@ -40,13 +40,13 @@ Accumulo has some optional native code that improves its performance and
 Run the command below to create configuration for Accumulo in `conf/`:
 
-    accumulo create-config
+    ./contrib/accumulo-config
```
This was moved to `bin/accumulo-util create-config`

mikewalch commented on assemble/conf/examples/generic_logger.properties:
```diff
@@ -15,7 +15,7 @@
 # Write out everything at the DEBUG level to the debug log
 log4j.appender.A2=org.apache.log4j.RollingFileAppender
-log4j.appender.A2.File=${org.apache.accumulo.core.dir.log}/${org.apache.accumulo.core.application}_${accumulo.service.instance}${org.apache.accumulo.core.ip.localhost.hostname}_fromprops.debug.log
+log4j.appender.A2.File=${org.apache.accumulo.core.dir.log}/${org.apache.accumulo.core.application}${accumulo.service.instance}_${org.apache.accumulo.core.ip.localhost.hostname}_fromprops.debug.log
```
This should be done in a separate PR.  Created issue https://issues.apache.org/jira/browse/ACCUMULO-4588

mikewalch commented on assemble/conf/examples/generic_logger.properties:
```diff
@@ -24,7 +24,7 @@ log4j.appender.A2.layout.ConversionPattern=%d{ISO8601} [%-8c{2}] %-5p: %m%n
 # Write out INFO and higher to the regular log
 log4j.appender.A3=org.apache.log4j.RollingFileAppender
-log4j.appender.A3.File=${org.apache.accumulo.core.dir.log}/${org.apache.accumulo.core.application}_${accumulo.service.instance}${org.apache.accumulo.core.ip.localhost.hostname}_fromprops.log
+log4j.appender.A3.File=${org.apache.accumulo.core.dir.log}/${org.apache.accumulo.core.application}${accumulo.service.instance}_${org.apache.accumulo.core.ip.localhost.hostname}_fromprops.log
```
This should be done in a separate PR.  Created issue https://issues.apache.org/jira/browse/ACCUMULO-4588

mikewalch commented on INSTALL.md:
```diff
@@ -76,19 +76,18 @@ When configuring Accumulo the following information about these dependencies
  * **Location of Zoookeeper and Hadoop jars** :  Setting `ZOOKEEPER_HOME` and
-   `HADOOP_PREFIX` in `conf/accumulo-env.sh` will help Accumulo find these
-   jars.
+   `HADOOP_PREFIX` in `accumulo.conf` will help Accumulo find these jars.
```
Improve documentation in 1e8d87cd311

mikewalch commented:
Thanks everyone for reviewing!  Your feedback resulted in a lot of changes that made this PR better.  I think I have addressed all comments.  Unless someone needs more time to review or wants more changes, I plan on squashing and merging by end of today.

mikewalch commented:
I talked to @milleruntime offline about accumulo-env.sh rename.  For this PR, I decided to not do the rename as concerns were brought up about changing name to accumulo.conf.  In the end, I would like a better name for accumulo-env.sh but I don't want to force a change until we find a better name.
